README.md

```diff
@@ -10,58 +10,35 @@ We developed J2V8 as a high performance engine for our multi-platform mobile too
 
 Building J2V8
 =============
-Building J2V8 requires building both the native parts and the Java library (.jar file). To build the native parts we first build node.js as a library and then statically link J2V8 to that. The Java parts are built with maven.
-
-Building on MacOS
------------------
-```
- sh ./build-node.sh
- sh ./buildJ2V8_macos.sh
- mvn clean verify
-```
-
-Building on Linux
------------------
-```
-export CCFLAGS="${CCFLAGS} -fPIC"
-export CXXFLAGS="${CXXFLAGS} -fPIC"
-export CPPFLAGS="${CPPFLAGS} -fPIC"
-#sh ./build-node.sh
-cp -r /data/jenkins/node .
-cd jni
-g++ -I../node -I../node/deps/v8 -I../node/deps/v8/include \
-    -I../node/src -I /data/jenkins/tools/hudson.model.JDK/jdk-7/include/ \
-    -I /data/jenkins/tools/hudson.model.JDK/jdk-7/include/linux  \
-    com_eclipsesource_v8_V8Impl.cpp -std=c++11 -fPIC -shared -o libj2v8_linux_x86_64.so \
-    -Wl,--whole-archive ../node/out/Release/libnode.a  -Wl,--no-whole-archive \
-    -Wl,--start-group \
-                      ../node/out/Release/libv8_libbase.a \
-                      ../node/out/Release/libv8_libplatform.a \
-                      ../node/out/Release/libv8_base.a \
-                      ../node/out/Release/libv8_nosnapshot.a \
-                      ../node/out/Release/libuv.a \
-                      ../node/out/Release/libopenssl.a \
-                      ../node/out/Release/libhttp_parser.a \
-                      ../node/out/Release/libgtest.a \
-                      ../node/out/Release/libzlib.a \
-                      ../node/out/Release/libcares.a \
-    -Wl,--end-group \
-    -lrt -D NODE_COMPATIBLE=1
-mvn clean verify
-```
-
-This will build J2V8 with node.js support. To disable this support, remove the `-D NODE_COMPATIBLE=1` option.
-
-Building for Android
------------------
-Building J2V8 for Android requires Docker.
-
-```
-./docker/build.sh
-./gradlew assembleRelease
-```
-
-This will build J2V8 as an AAR for API 19 minimum with node.js support.
+Building J2V8 requires building both the native parts and the Java library (.jar/.aar file). To build the native parts we first build node.js as a library and then statically link J2V8 to that. The Java parts are built with maven/gradle.
+
+J2V8 uses a cross-platform, cross-compiling build-system written in Python.
+
+Follow these steps to build J2V8 from source:
+
+1) clone the Node.js source code
+    - `python prepare_build.py`
```

Comment:
even most simple build didn't work on my latest manjaro (arch) linux:
python2 build.py -t linux -a x64

most likely I ran into this issue: gcc7 doesn't compile:

```
make[1]: *** [deps/v8/src/v8_base.target.mk:554: /home/mati/Devel/OpenSource/J2V8/node/out/Release/obj.target/v8_base/deps/v8/src/heap/heap.o] Error 1
make[1]: *** [deps/v8/src/v8_base.target.mk:554: /home/mati/Devel/OpenSource/J2V8/node/out/Release/obj.target/v8_base/deps/v8/src/heap/mark-compact.o] Error 1
make[1]: Leaving directory '/home/mati/Devel/OpenSource/J2V8/node/out'
make: *** [Makefile:73: node] Error 2
Traceback (most recent call last):
  File "build.py", line 311, in <module>
    execute_build(args)
  File "build.py", line 307, in execute_build
    execute_build_step(target_compiler, target_step)
  File "build.py", line 266, in execute_build_step
    compiler.build(build_step)
  File "/home/mati/Devel/OpenSource/J2V8/build_system/cross_build.py", line 60, in build
    self.exec_build(config)
  File "/home/mati/Devel/OpenSource/J2V8/build_system/shell_build.py", line 26, in exec_build
    self.exec_cmd(shell_str, config)
  File "/home/mati/Devel/OpenSource/J2V8/build_system/cross_build.py", line 79, in exec_cmd
    utils.execute(cmd, dir)
  File "/home/mati/Devel/OpenSource/J2V8/build_system/build_utils.py", line 37, in execute
    raise subprocess.CalledProcessError(return_code, cmd)
subprocess.CalledProcessError: Command 'cd /home/mati/Devel/OpenSource/J2V8 && cd ./node && ./configure                          --without-intl                      --without-inspector                 --dest-cpu=x64                    --without-snapshot                  --enable-static && CFLAGS=-fPIC CXXFLAGS=-fPIC make -j4' returned non-zero exit status 2
```
https://github.com/nodejs/help/issues/664

this makes me think that perhaps docker should always be used...